Vizzuality/mangrove-atlas
# fixes and updates eslint dependencies

This PR adds `eslint` (it wasn't added as a dev dependency, best case it would use the one installed locally on your machine, which is not ideal), updates its dependencies and add some new ones.

I haven't run the linter to fix all errors as this PR would grow a lot. Feel free to run it in another PR or just fix them as you go. Also, I have added some rules to avoid changing the way components are declared and some additional rules, but feel free to change them if needed.

To test:
- run `rm -rf node_modules && yarn` in your terminal.
- go to some file, if there are errors/warnings it should display on your screen.

PS: I also recommend adding to your VS Code configuration the following lines to run eslint on save:

```
"editor.codeActionsOnSave": {
    "source.fixAll.eslint": true
}
```
